scripts/quests/windurst/Heaven_Cent.lua

```diff
@@ -0,0 +1,274 @@
+-----------------------------------
+ID = require('scripts/zones/Maze_of_Shakhrami/IDs')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.HEAVEN_CENT)
+
+local incorrect =
+{
+    ID.text.SHIVA_WEST,
+    ID.text.LEAVIATHAN_SOUTH,
+    ID.text.TITAN_NORTH,
+    ID.text.ODIN_EAST,
+}
+
+local correct =
+{
+    ID.text.ALEXANDER_NE,
+    ID.text.SHIVA_EAST,
+    ID.text.ODIN_NORTH,
+    ID.text.IFRIT_NW,
+}
+
+quest.reward =
+{
+    gil = 4800,
+    title = xi.title.NIGHT_SKY_NAVIGATOR,
+    fame = 100,
+    fameArea = xi.quest.fame_area.WINDURST,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:getFameLevel(xi.quest.fame_area.WINDURST) >= 5
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ropunono'] = quest:progressEvent(284),
+
+            onEventFinish =
+            {
+                [284] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ropunono'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(289, 0, 557, 545, 0, 0, 4095, 4)
+                    else
+                        return quest:progressEvent(296)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if trade:hasItemQty(557, 1) and trade:getItemCount() == 1 and quest:getVar(player, 'Prog') == 0 then
+                        return quest:progressEvent(288, 0, 557, 545, 0, 0, 0, 0, 4)
+
+                    -- Fake Shelling Piece
+                    elseif
+                        trade:hasItemQty(545, 1) and
+                        trade:getItemCount() == 1 and
+                        quest:getVar(player, 'Prog') == 1 and
+                        quest:getVar(player, 'Coin') == 0
+                    then
+                        return quest:progressEvent(296, 0, 0, 545, 0, 0, 0, 4095, 4)
+
+                    -- Real Shelling Piece
+                    elseif
+                        trade:hasItemQty(545, 1) and
+                        trade:getItemCount() == 1 and
+                        quest:getVar(player, 'Prog') == 1 and
+                        quest:getVar(player, 'Coin') == 1
+                    then
+                        return quest:progressEvent(292, 0, 0, 545, 0, 0, 0, 4095, 4)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [288] = function(player, csid, option, npc)
+                    quest:setvar(player, 'Prog', 1)
```

Comment:
Lua functions are case sensitive; `setvar` will be nil compared to `setVar`

Reply:
Ope! Thanks bud! Think I fixed this in my local. But I'll go through and make sure I got them all.